File: WebApplication/WebApplication/Models/Content/TermViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApplication.Models.Content
{
    public class TermViewModel
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Source { get; set; }
        public List<IdValueModel> Themes { get; set; }
        public List<TermSimpleModel> RelatedTerms { get; set; }
    }
}

File: WebApplication/WebApplication/ClientApp/src/models/content/term-view.model.ts
import { IdValueModel } from "../id-value.model";
import { TermSimpleModel } from "./term-simple.model";

export class TermViewModel {
  public id: number;
  public name: string;
  public description: string;
  public source: string;
  public themes: IdValueModel[];
  public relatedTerms: TermSimpleModel[];

  //additional properties
  public is404: boolean = false;
  public isLoading: boolean = false;

}

export const TERM_VIEW_404 = { is404: true } as TermViewModel;
export const TERM_VIEW_LOADING = { isLoading: true } as TermViewModel;
